plugins/de.cau.cs.kieler.spviz.spviz/src/de/cau/cs/kieler/spviz/spviz/generator/GenerateSubSyntheses.xtend

```diff
@@ -0,0 +1,368 @@
+							children += «overview.toFirstLower»OverviewContext.detailed«artifact»Contexts.flatMap [
+								return «artifact.toFirstLower»Synthesis.transform(it as «artifact»Context)
+							]
+						«ENDFOR»
+						
+						«FOR connection : spviz.getOverviewConnections(overview)»
+						// Add all by «connection.get(1).toFirstLower»s required «connection.get(2).toFirstLower»s edges.
+						«overview.toFirstLower»OverviewContext.required«connection.get(0)»«connection.get(2)»Edges.forEach [
+«««							// Connects the {@code sourceBundleNode} and the {@code usedByBundleNode} via an arrow in UML style,
+«««							// so [usedByBundleNode] ----- uses -----> [sourceBundleNode]
+							val requiring = key
+							val required = value
+							if (!nodeExists(requiring) || !nodeExists(required)) {
+								// Only Add edges if the nodes are actually shown.
+								return
+							}
+							val requiringNode = requiring.node
+							val requiredNode = required.node
+							val requiringPort = requiringNode.ports.findFirst [
+								data.filter(KIdentifier).head?.id === "required«connection.get(0)»«connection.get(2)»s"
+							]
+							val requiredPort = requiredNode.ports.findFirst [
+								data.filter(KIdentifier).head?.id === "requiring«connection.get(0)»«connection.get(2)»s"
+							]
+							
+							val edge = createEdge(requiring, required) => [
+								addRequired«connection.get(0)»«connection.get(2)»EdgeRendering
+								sourcePort = requiringPort
+								targetPort = requiredPort
+								source = requiringNode
+								target = requiredNode
+							]
+							requiringNode.outgoingEdges += edge
+							requiredNode.incomingEdges += edge
+						]
+						«ENDFOR»
+					]
+				}
+				
+			}
+			
+		'''
+	}
+	
+	/**
+	 * Generates the syntheses contend for a given artifact.
+	 * 
+	 * @param artifact
+	 * 		the artifact name as a string
+	 * @param spviz
+	 * 		a DataAccess to easily get the information from
+	 * @return 
+	 * 		the generated file content as a string
+	 */
+	def static String generateSynthesis(String artifact, DataAccess spviz){
+		val overview = spviz.getOverview(artifact)
+		return '''
+			package «spviz.packageName».viz.subsyntheses
+			
+			import com.google.inject.Inject
+			import de.cau.cs.kieler.klighd.kgraph.KGraphFactory
+			import de.cau.cs.kieler.klighd.kgraph.KNode
+			import de.cau.cs.kieler.klighd.krendering.extensions.KNodeExtensions
+			import de.cau.cs.kieler.klighd.krendering.extensions.KPortExtensions
+			import de.cau.cs.kieler.klighd.syntheses.AbstractSubSynthesis
+			import «spviz.importedNamespace».model.«artifact»
+			«FOR required : spviz.getRequiredArtifacts(artifact)»
+				«IF required.get(1) != artifact»
+					import «spviz.importedNamespace».model.«required.get(1)»
+				«ENDIF»
+			«ENDFOR»
+			«FOR requiring : spviz.getRequiredArtifacts(artifact)»
+				«IF requiring.get(1) != artifact»
+					import «spviz.importedNamespace».model.«requiring.get(1)»
+				«ENDIF»
+			«ENDFOR»
```

Comment:
Does this also cover imports if required and requiring are the same or are they duplicated then?
I think using a `Set` here, putting in all needed imports and getting all non-duplicates from that set would be better as it always cover all cases and takes care of the logic for us.